lib/OpenQA/WebAPI.pm

```diff
@@ -74,6 +74,29 @@ sub startup {
     # Set some application defaults
     $self->defaults(appname => $self->app->config->{global}->{appname});
 
+    # Get application version
+    if (-e $self->home->child('.git') and -e $self->home->child('.git/refs/heads/master')) {
+        my $version = file_content($self->home->child('.git/refs/heads/master'));
+
+        # Extrapolate latest tagged version and combine it
+        # with latest commit which heads is pointed to.
+        # This method have its limits while checking out different branches
+        # but emulates git-describe output without executing commands.
+        my $packed_refs = file_content($self->home->child('.git/packed-refs'));
+        if ($packed_refs) {
+            my $latest_ref = (grep(/tags/, split(/\s/, $packed_refs)))[-1];
+            if ($latest_ref) {
+                my $latest_tag = (split(/\//, $latest_ref))[-1];
+                $self->defaults(current_version => "git-" . $latest_tag . "-" . substr($version, 0, 8))
+                  unless !$latest_tag;
```

Comment:
do not use post conditions - if the line is longer, especially if tidy breaks the line.

And do not use unless !